src/ShapeletFunction.cc

```diff
@@ -169,13 +169,14 @@ ShapeletFunctionEvaluator::ShapeletFunctionEvaluator(
 ShapeletFunction ShapeletFunction::convolve(ShapeletFunction const & other) const {
     GaussHermiteConvolution convolution(getOrder(), other);
     afw::geom::ellipses::Ellipse newEllipse(_ellipse);
-    auto matrix = convolution.evaluate(newEllipse).asEigen();
+    auto matrixNdArray = convolution.evaluate(newEllipse);
+    auto matrix = ndarray::asEigenMatrix(matrixNdArray);
```

Comment:
Consider adding a comment saying that `matrixNdArray` must be a local variable, to prevent accidental refactoring. Though since `matrix` is only used at one point, you could also just do the evaluation and conversion there.

Reply:
Good point. I'll eliminate the `matrix` temporary.